filament/backend/src/vulkan/VulkanUtility.cpp

```diff
@@ -528,6 +529,30 @@ bool equivalent(const VkExtent2D& a, const VkExtent2D& b) {
     return a.height == b.height && a.width == b.width;
 }
 
+bool isDepthFormat(VkFormat format) {
+    switch (format) {
+        case VK_FORMAT_D16_UNORM:
+        case VK_FORMAT_X8_D24_UNORM_PACK32:
+        case VK_FORMAT_D16_UNORM_S8_UINT:
+        case VK_FORMAT_D24_UNORM_S8_UINT:
+        case VK_FORMAT_D32_SFLOAT:
+        case VK_FORMAT_D32_SFLOAT_S8_UINT:
+            return true;
+        default:
+            return false;
+    }
+}
+
+static uint32_t mostSignificantBit(uint32_t x) { return 1ul << (31ul - utils::clz(x)); }
+
+uint8_t reduceSampleCount(uint8_t sampleCount, VkSampleCountFlags mask) {
+    assert_invariant(utils::popcount(sampleCount) == 1);
+    if (sampleCount & mask) {
+        return sampleCount;
+    }
+    return mostSignificantBit((sampleCount - 1) & mask);
```

Comment:
This is how I was originally planning on doing it too.. I have this exact expression written on my notepad lol